src/protagonist/API.Tests/Features/CustomHeaders/Validation/HydraCustomHeaderValidator.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using API.Features.CustomHeaders.Validation;
+using API.Features.CustomHeaders.Converters;
+using DLCS.HydraModel;
+using FluentValidation.TestHelper;
+
+namespace API.Tests.Features.CustomHeaders.Validation;
+
+public class HydraCustomHeaderValidatorTests
+{
+    private readonly HydraCustomHeaderValidator sut;
+    
+    public HydraCustomHeaderValidatorTests()
+    {
+        sut = new HydraCustomHeaderValidator();
+    }
+    
+    [Fact]
+    public void NewCustomHeader_CannotHave_AssetId()
+    {
+        var customHeader = new CustomHeader()
+        {
+            Id = Guid.NewGuid().ToString()
+        };
+        var result = sut.TestValidate(customHeader);
+        result.ShouldHaveValidationErrorFor(ch => ch.Id);
+    }
+    
+    [Fact]
+    public void NewCustomHeader_CannotHave_CustomerId()
+    {
+        var customHeader = new CustomHeader()
+        {
+            CustomerId = 1
+        };
+        var result = sut.TestValidate(customHeader);
+        result.ShouldHaveValidationErrorFor(ch => ch.CustomerId);
+    }
+    
+    [Fact]
+    public void NewCustomHeader_Requires_Key()
+    {
+        var customHeader = new CustomHeader();
+        var result = sut.TestValidate(customHeader);
+        result.ShouldHaveValidationErrorFor(ch => ch.Key);
+    }
+    
+    [Fact]
+    public void NewCustomHeader_Requires_Value()
+    {
+        var customHeader = new CustomHeader();
+        var result = sut.TestValidate(customHeader);
+        result.ShouldHaveValidationErrorFor(ch => ch.Value);
```

Comment:
might be worth validating that the values you haven't set are null in this set of tests?

Reply:
Good idea. Added in df51740576117eeabff930b0a854996b3ed7c0aa